packages/vhd-lib/src/Vhd/VhdDirectory.js

```diff
@@ -0,0 +1,195 @@
+import { BLOCK_UNUSED } from '../../dist/_constants'
+import { createLogger } from '@xen-orchestra/log'
+import { fuFooter, fuHeader, checksumStruct } from '../_structs'
+import { SECTOR_SIZE } from '../_constants'
+import { sectorsRoundUpNoZero, sectorsToBytes, buildHeader, buildFooter, computeBatSize } from './_utils'
+import { test, set as setBitmap } from '../_bitmap'
+import { VhdAbstract } from './VhdAbstract'
+import assert from 'assert'
+
+const { debug } = createLogger('vhd-lib:ChunkedVhd')
+
+export class VhdDirectory extends VhdAbstract {
+  static async open(handler, path, flags) {
```

Comment:
Do you think we should create separate methods to open an existing VHD and create a new one?

We could remove the flags and make the API easier to use.

Reply:
The overwrite checks are only for the `copy` command, maybe we can remove this limit

I am more concerned about the acrobatics needed to open a Vhd if we don't know if it's a directory one or a file one.

Reply:
maybe, taking the open function up to AbstractVhd or something like a Vhd factory that will use the right Vhd subclass

Reply:
Maybe opening an existing VHD should automatically read header and footer?

Reply:
it will fail when copying to an empty location. But for any other usage, I think reading header, footer (and maybe block allocation table ?)  during `open` should not be too expensive and will make the usage easier